src/Swashbuckle.AspNetCore.Swagger/SwaggerMiddleware.cs

```diff
@@ -15,19 +18,29 @@ public class SwaggerMiddleware
         private readonly RequestDelegate _next;
         private readonly SwaggerOptions _options;
         private readonly TemplateMatcher _requestMatcher;
+#if !NETSTANDARD
+        private readonly TemplateBinder _templateBinder;
+#endif
 
         public SwaggerMiddleware(
             RequestDelegate next,
-            SwaggerOptions options)
+            SwaggerOptions options
+#if !NETSTANDARD
+            ,TemplateBinderFactory templateBinderFactory
+#endif
+            )
         {
             _next = next;
             _options = options ?? new SwaggerOptions();
             _requestMatcher = new TemplateMatcher(TemplateParser.Parse(_options.RouteTemplate), new RouteValueDictionary());
+#if !NETSTANDARD
+            _templateBinder = templateBinderFactory.Create(RoutePatternFactory.Parse(_options.RouteTemplate));
```

Comment:
Is there any danger that someone's existing custom `RouteTemplate` would start throwing an exception when passed through this code?

Reply:
It would throw the line before in `TemplateParser.Parse(_options.RouteTemplate)` which uses `RoutePatternFactory.Parse` under the hood. Anyway, you'd want an invalid route template to throw.